beamer/chain.py

```diff
@@ -225,6 +226,15 @@ def process_requests(context: Context) -> None:
     for request in context.requests:
         log.debug("Processing request", request=request)
 
+        if request.l1_resolution_tx_handle is not None:
+            handle: Future = request.l1_resolution_tx_handle
+
+            if handle.done():
+                try:
+                    handle.result()
+                except Exception as ex:
+                    log.warning("L1 Resolution failed", ex=ex)
+
```

Comment:
There are a couple of things here:
- I think this should be moved outside of this function, preferably into its own function which would then be called from the caller of this one (this block does not really relate to request processing)
- why do we catch all exceptions here?
- why do we log exceptions via `warning` instead of `error`?

Reply:
1) An alternative I see is putting all the futures in a separate queue and poll theme there. This would also remove them from the `Request` objects for a simple boolean flag instead. Opinions on that?
2) `result` forwards all exceptions the internal call might yield. So this is the safe thing to do here.

Reply:
> 1. An alternative I see is putting all the futures in a separate queue and poll theme there. This would also remove them from the `Request` objects for a simple boolean flag instead. Opinions on that?

Hmm... not a bad idea. But then I would go a step further and introduce a new thing that manages L1 resolutions,
e.g. `L1ResolutionManager` or something like that, which would be common to all (future) chains.
The thing would run in its own thread, like the event monitors and the even processor we already have.
Then it would take care of all things related to L1 resolution and you would merely need to track the state on request,
which would be in line with our state machine vision.

What do you think?

>     2. `result` forwards all exceptions the internal call might yield. So this is the safe thing to do here.

Yes, but I'm wondering why do we catch _all_ exceptions here?
If an exception happens, we would like to be notified about it by shutting down the agent. This way, it will be buried in the logs.

Reply:
> Hmm... not a bad idea. But then I would go a step further and introduce a new thing that manages L1 resolutions,
> e.g. `L1ResolutionManager` or something like that, which would be common to all (future) chains.
> The thing would run in its own thread, like the event monitors and the even processor we already have.
> Then it would take care of all things related to L1 resolution and you would merely need to track the state on request,
> which would be in line with our state machine vision.

I see the idea, but think it would be mostly a simple wrapper around the `ThreadPoolExecutor` anyways, so I'd rather avoid it. The whole L1 resolution is nothing more than a list of futures, and my expectation is that the futures will be short-lived as well.